File: backend/src/main/kotlin/org/qask/backend/models/viewModels/QuestionPayload.kt
package org.qask.backend.models.viewModels

import org.qask.backend.models.enums.Status
import java.time.Instant

// used in streaming
data class QuestionPayload(
    val content: String? = "",
    val questionerName: String? = null,
    val likeCount: Int = 0,
    val sent: Instant,
    val isDeleted: Boolean = false,
    val isHidden: Boolean = false,
    val id: String? = null,
    val status: String = Status.DEFAULT.name
)

File: frontend/src/types/QuestionPayload.ts
import Status from "./Status";

interface QuestionPayload {
    content: string,
    questionerName: string,
    likeCount: number,
    sent: string,
    isDeleted: boolean,
    isInsincere: boolean,
    status: Status
    id: string,
}

export default QuestionPayload;